src/maths/calculator/calculatorComponent.js

```diff
@@ -46,70 +43,64 @@ var defaultConfig = {
          * Gets the nested calculator
          * @returns {calculator}
          */
-        getCalculator: function getCalculator() {
+        getCalculator() {
             return calculator;
         }
     };
 
-    calculatorComponent = dynamicComponent(api, defaultConfig)
-        .on('rendercontent', function($content) {
-            const self = this;
-            const initialWidth = self.getElement().width();
-            const initialHeight = self.getElement().height();
+    const calculatorComponent = dynamicComponent(api, defaultConfig)
+        .on('rendercontent', function onRenderContent($content) {
+            const initialWidth = this.getElement().width();
+            const initialHeight = this.getElement().height();
 
-            return pluginsLoader(this.getConfig().loadedPlugins, this.getConfig().dynamicPlugins).then(function(
-                loadedPlugins
-            ) {
-                return new Promise(function(resolve) {
-                    calculator = calculatorBoardFactory($content, loadedPlugins, self.getConfig().calculator).on(
-                        'ready',
-                        function() {
-                            var initialFontSize =
-                                parseInt(
-                                    self
-                                        .getCalculator()
-                                        .getElement()
-                                        .css('fontSize'),
-                                    10
-                                ) || 10;
-                            self.on('resize', function() {
-                                if (self.getElement()) {
-                                    self.getCalculator()
-                                        .getElement()
-                                        .css(
-                                            'fontSize',
-                                            initialFontSize *
-                                                Math.min(
-                                                    self.getElement().width() / initialWidth,
-                                                    self.getElement().height() / initialHeight
-                                                )
-                                        );
-                                }
-                            })
-                                .setContentSize(
-                                    calculator.getElement().outerWidth(),
-                                    calculator.getElement().outerHeight()
-                                )
-                                .setState('ready')
-                                .trigger('ready');
-                            resolve();
-                        }
-                    );
+            return pluginLoaderFactory(this.getConfig().loadedPlugins)
+                .addList(this.getConfig().dynamicPlugins)
+                .load(context.bundle)
+                .then(loadedPlugins => {
```

Comment:
**note:** The plugin loader can be simplified since most of the plugins have been removed as they are directly implemented by the engine.